src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.TryCloneFile.OSX.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        private static partial bool TryCloneFile(string sourceFullPath, in Interop.Sys.FileStatus srcStat, string destFullPath, bool overwrite)
+        {
+            // Try to clone the file immediately, this will only succeed if the destination doesn't exist,
+            // so we don't worry about locking for this one.
+            if (Interop.@libc.clonefile(sourceFullPath, destFullPath, Interop.@libc.CLONE_ACL) == 0)
+            {
+                // Success
+                return true;
+            }
+            Interop.Error error = Interop.Sys.GetLastError();
+
+            // Try to delete the destination file if we're overwriting.
+            if (error == Interop.Error.EEXIST && overwrite)
+            {
+                // Get the file permissions for source file
+                UnixFileMode filePermissions = (UnixFileMode)srcStat.Mode & SafeFileHandle.PermissionMask;
```

Comment:
We don't need a `filePermissions` for opening the destination handle. The permissions are only used when creating a new file. The handle is for an existing file (`FileMode.Open`)